dipy/denoise/patch2self.py

```diff
@@ -158,6 +149,204 @@ def _extract_3d_patches(arr, patch_radius):
     return np.array(all_patches).T
 
 
+def _fit_denoising_model(train, vol_idx, model, alpha, version):
+    """Fit a single 3D volume using a train and test phase.
+
+    Parameters
+    ----------
+    train : ndarray
+        Array of all 3D patches flattened out to be 2D.
+
+    vol_idx : int
+        The volume number that needs to be held out for training.
+
+    model : str or initialized linear model object.
+        This will determine the algorithm used to solve the set of linear
+        equations underlying this model. If it is a string it needs to be
+        one of the following: {'ols', 'ridge', 'lasso'}. Otherwise,
+        it can be an object that inherits from
+        `dipy.optimize.SKLearnLinearSolver` or an object with a similar
+        interface from Scikit-Learn:
+        `sklearn.linear_model.LinearRegression`,
+        `sklearn.linear_model.Lasso` or `sklearn.linear_model.Ridge`
+        and other objects that inherit from `sklearn.base.RegressorMixin`.
+        Default: 'ridge'.
+
+    alpha : float, optional
```

Comment:
There is no default value in the method signature, so this should not be marked optional.

Reply:
Hello @jhlegarreta. Thank you so much for the review. I'm really sorry about the commit messages, I was putting the commits as just fixes because I was trying to test out the CIs. I will try to keep it as draft next time.
I was also planning on squashing all the commits towards the end. Sry about the confusing commits once again.

I'll also recheck on all the function docstrings once again. Thank you for your inputs.